sphinx/ext/autosummary/generate.py

```diff
@@ -188,6 +189,29 @@ def get_members(obj, typ, include_public=[], imported=True):
                           if x in include_public or not x.startswith('_')]
                 return public, items
 
+            def get_package_members(out_dict, include_public=[], imported=True):
+                # https://docs.python.org/3/reference/import.html#packages
+                # ... any module that contains a __path__ attribute is
+                # considered a package
+                if not hasattr(obj, '__path__'):
+                    return
+                out_dict['modules'], out_dict['all_modules'] = [], []
+                out_dict['packages'], out_dict['all_packages'] = [], []
```

Comment:
When are `all_modules` and `all_packages` used?

Reply:
Analog to `all_functions`, `all_classes`, `all_methods`, etc... I suppose they are only used when you specify them in your template.

Reply:
Reasonable.